pyro/params/__init__.py

```diff
@@ -1,23 +1,12 @@
 from __future__ import absolute_import, division, print_function
 
-from .param_store import ParamStoreDict
-
-# the global ParamStore
-_PYRO_PARAM_STORE = ParamStoreDict()
-
-# used to create fully-formed param names, e.g. mymodule$$$mysubmodule.weight
-_MODULE_NAMESPACE_DIVIDER = "$$$"
-
-
-def param_with_module_name(pyro_name, param_name):
-    return _MODULE_NAMESPACE_DIVIDER.join([pyro_name, param_name])
-
-
-def module_from_param_with_module_name(param_name):
-    return param_name.split(_MODULE_NAMESPACE_DIVIDER)[0]
-
-
-def user_param_name(param_name):
-    if _MODULE_NAMESPACE_DIVIDER in param_name:
-        return param_name.split(_MODULE_NAMESPACE_DIVIDER)[1]
-    return param_name
+from pyro.params.param_store import _MODULE_NAMESPACE_DIVIDER, _PYRO_PARAM_STORE, \
+    module_from_param_with_module_name, param_with_module_name, user_param_name
+
+__all__ = [
+    "_MODULE_NAMESPACE_DIVIDER",
+    "_PYRO_PARAM_STORE",
```

Comment:
nit: this isnt relevant to this PR per se, but if these are being exposed or used outside the module then they shouldnt have `_` prepended?

Reply:
+1, agreed. Let me see if we can remove this, and just use the get functions everywhere.

Reply:
Hmm I consider `_` to denote pravacy wrt the repo, rather than privacy wrt a module. E.g. PyTorch uses `_` to denote internal methods with no backwards-compatibility guarantees. Therefore,
1. it should be fine to import underscored names across modules;
2. we don't need to add undescored names to `__all__`, since `__all__` only determins `*` import, and we avoid `*` import internally.

Reply:
That makes sense, will clean that up shortly.